roles/ceph-ansible/tasks/main.yml

```diff
@@ -100,16 +100,46 @@
     regexp: "^#journal_size: 0.*"
     line: 'journal_size: 5120'
 
-- name: calculate network prefix
-  shell: "ipcalc -p 0.0.0.0 {{ hostvars[groups['mons'][0]].ansible_default_ipv4['netmask'] }} | sed 's|PREFIX=||'"
-  register: net_prefix
-  changed_when: false
+- set_fact:
+    pub_network: "{{ hostvars[groups['mons'][0]].ansible_default_ipv4.network }}"
+    pub_prefix: "{{ ['0.0.0.0/', hostvars[groups['mons'][0]].ansible_default_ipv4.netmask] | join('') | ipaddr('prefix') }}"
 
-- name: Set public_network to ansible_default_ipv4['address']/16
+- name: Set public_network to ansible_default_ipv4['network']/prefix
   lineinfile:
     dest: "{{ base_dir }}/ceph-ansible/group_vars/all.yml"
     regexp: "^#public_network: 0.0.0.0/0.*"
-    line: "public_network: {{ hostvars[groups['mons'][0]].ansible_default_ipv4['network'] }}/{{ net_prefix.stdout }}"
+    line: "public_network: {{ pub_network }}/{{ pub_prefix }}"
+
+- name: Filter additional NICs (exclude ansible default interface and lo)
+  set_fact:
+    additional_network_interfaces: "{{ hostvars[groups['osds'][0]].ansible_interfaces | difference(['lo', hostvars[groups['osds'][0]].ansible_default_ipv4.interface]) }}"
```

Comment:
I see that you use `groups['osds'][0]` instead of previously used `groups['mons'][0]` to get the default interface. Is this intentional or would it be better to use the same expression to get it in both cases?